2017/proposals/bansal-utkarsh.md

```diff
@@ -0,0 +1,628 @@
+fixtures. This part should go relatively smoothly because I will be able to follow the patterns and conventions
+developed while working on the previous part of the project. Again, PR's will be made on a per file basis.
+
+
+The following modules will be fixed in this part -
+* `analysis` (21 files)
+* `auxiliary` (2 files)
+* `core` (17 files)
+* `formats` (1 file)
+* `lib` (1 file)
+* `topology` (20 files)
+* Files in the root of `testsuite` directory (19 files)
+
+
+I want to work on parts 2 and 3 on a per-file basis because I find small changes are easy to review and get merged. Also
+this way, I will be able to stick to my proposed timeline. This will also help my mentors to easily keep track of my progress
+and help me sooner, if I were to start lagging behind or go off track.
+
+
+Other Research
+--------------
+Just to get an approximate how much time _part 1_ of this project would take, I went ahead and did a small part of it on
+my fork. I worked on the `coordinates` module (which is also the largest module) and made some changes to make it work
+with pytest. Here are my findings
+
+**Change #1** Changed all classes that have either setUp or tearDown methods to inherit from TestCase.
+
+**Before** `580 failed, 954 passed, 31 skipped, 505 pytest-warnings, 26 error in 146.41 seconds`
+
+**After**  `171 failed, 1361 passed, 87 skipped, 42 pytest-warnings, 50 error in 177.44 seconds`
+
+In short, there are around 1600 test cases, by changing the base class I got around 400 more to run successfully with pytest.
+
+**Change #2** Use `__test__` attribute in classes which are either inheriting from other base test classes & classes that
+are being inherited from i.e. both parent and child classes.
+
+**Before** `171 failed, 1361 passed, 87 skipped, 42 pytest-warnings, 50 error in 177.44 seconds`
+
+**After**  `1408 passed, 68 skipped`
+
+In short, _none of the test cases fail_.
+
+It is to be noted that pytest collects around 270 test cases less that nosetests, that is largely because of the fact that
+pytest does not collect classes with an `__init__` method (which I've explained above)
+
+Also to get an approximate of how much time and effort _part 2_ would take, I have worked on a part of issue [#516](https://github.com/MDAnalysis/mdanalysis/issues/516)
+where I fixed the `GRO` writers API to follow the reader API standard and modified test classes to inherit from base test classes. As of now the changes have been
+approved and are ready to be merged. See [#1196](https://github.com/MDAnalysis/mdanalysis/pull/1196)
+
+
+Schedule of Deliverables
+========================
+
+### May 4th - May 29th, **Community Bonding Period**
+
+### May 30th - June 3rd
+* Begin work on **Part 1**. Start by editing tests to make them discoverable by pytest.
+Also actively discuss the `__init__` issue as it will be dealt with on a case by case
+basis.
+* Send a PR only for review and discussion purposes. It will _not_ be ready to merge.
```

Comment:
Plugins are important, but less crucial. We can do without it for a short while. We are actually doing without the `open_files` plugin for long now.

So I think plugin can go in an other PR without much of a problem.

What we **really** do not want, id the develop branch to loose tests at any point.